### DISCUSSION TITLE
A couple of questions around CustomAppViews

### DISCUSSION THREAD

**Discussion by loalf** [kubeapps/kubeapps#4991] [Q&A] [answered]
Hello everyone,

first things first. Thank you so much for building this amazing tool, I'm loving it.

I can see that this new feature to render custom app views have been added recently and I love how extensible this is becoming. I have a couple of questions thought about its usage and I'd highly appreciate it if you could take the time to answer them 🙏 

1. Can you have different CustomAppViews for different charts (i.e. chart `a` is rendered by `CustomAppViewA` whereas chart `b` is rendered by `CustomAppViewB` or we have to put together a `customAppView` that will load one or the other depending on the helm chart

2. According to the documentation, this bit of configuration needs to be passed on the dashboard
```
customAppViews:
  - plugin: helm.packages
    name: apache
    repository: bitnam
```

But I'm struggling to understand what values `plugin` is supposed to accept. I'm not quite sure I understand what plugin means in this context.

Again, thank you so much for your time!!!

**Comment by antgamdia** [comment, marked as answer]
Hola Javier 👋 !

Thanks for your kind words, we really appreciate it! 

For 1), I'd have to have a closer look at this feature, as it was mainly implemented by @aanthonyrizzo, but I'd rather say it is not possible. This custom view is per plugin, not per package, but not 100% sure at the moment.

Perhaps we should clarify our docs with respect to 2), yep. The `plugin` is the plugin `name` field in the Plugins API ([source](https://github.com/vmware-tanzu/kubeapps/blob/main/cmd/kubeapps-apis/proto/kubeappsapis/core/plugins/v1alpha1/plugins.proto#L58)) . Particulary, you can have a look at the current plugin names in [this piece of code](https://github.com/vmware-tanzu/kubeapps/blob/main/dashboard/src/shared/utils.ts#L15-L19). In short , they are: `helm.packages`, `fluxv2.packages` and `kapp_controller.packages`.

Thanks again for your interest. Please feel free to reach out to us in the  [#kubeapps slack channel](https://kubernetes.slack.com/messages/kubeapps) and tell us your use case. We are looking for more [Kubeapps adopters ](https://github.com/vmware-tanzu/kubeapps/blob/main/ADOPTERS.md) out there!

**Reply by loalf** [reply]
Hola Antonio!

thanks for your quick response! 🙏 

I'll give it a go and see how far I can go using this new feature, it seems quite promising.

Then again, thanks a million for your comprehensive answer!
